doc/development.rst

```diff
@@ -470,6 +470,23 @@ To remove this limitation, we should have a common way to make files available
 to the exporter, possibly by generating a hash locally and rsyncing new files to
 the exporter.
 
+Graph Strategy Shortcuts
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently non-default paths in Graph Strategies are only followed when
+`transition()` is called with the keyword argument `via`.
+In order to simplify this and make Graph Strategies more powerful the Strategy
+itself should be able to decide which is the best path to follow.
+
+  - Remember states that were executed
+  - Introduce a speed rating for each state (default could be `1`)
+  - Add "states that must have been executed before" argument for each state
+    (default should be an empty list)
+
+The Graph Strategy could now determine the best path (based on the speed
+rating) that is already unlocked (all required states were executed before)
+when `transition()` is called.
```

Comment:
This would seem to require that the user transitions to some special unlock states in the correct order. That would go against the idea that the user should only specify the intended state and not need to know about how to get there.

Is the intention of this to have a way to skip repeated state transitions that take a long time?

Reply:
The idea is that the "unlock states" are part of the default path. Yes, skipping states is the main reason for this proposal.